src/DataStructures/Tensor/Expressions/TensorExpression.hpp

```diff
@@ -312,7 +318,15 @@ namespace detail {
 /// `max_num_ops_in_sub_expression_helper`.
 ///
 /// Before defining a max operations cap for some data type, the change should
-/// be first justified by benchmarking before and after introducing the new cap.
+/// first be justified by benchmarking many different tensor expressions before
+/// and after introducing the new cap. The optimal cap will likely be
+/// hardware-dependent, so fine-tuning this would ideally involve benchmarking
+/// on each hardware architecture and then controling the value based on the
+/// hardware.
+///
+/// The current value set for when the data type is `DataVector` was benchmarked
+/// by compiling with clang-10 Release and running on a CentOS Linux 7 system
+/// with a 64-bit x86 architecture.
```

Comment:
That still leaves around 15 years of different CPUs :) Could you specify the actual CPU? `cat /proc/cpuinfo` should work. You want the `model name`, eg `AMD Ryzen Threadripper 3970X 32-Core Processor` or `Intel(R) Core(TM) i5-4210U CPU @ 1.70GHz`

Reply:
I got the model name from that command, thanks! Would this change be fine?:
```
... by compiling with clang-10 Release and running on <model name>.
```

Or do you want me to keep the CentOS Linux 7 part?

Reply:
I don't think the OS matters so much as the CPU. So if you want to omit the OS, go ahead :)

Reply:
Squashed in this change